packages/web/src/components/comments/CommentForm.tsx

```diff
@@ -58,10 +59,13 @@ export const CommentForm = ({
   const [editComment] = useEditComment()
 
   const location = useLocation()
+  const previousLocation = usePrevious(location)
   useEffect(() => {
-    // Reset input text when the location changes
-    setMessageId((prev) => prev + 1)
-  }, [location])
+    if (previousLocation && previousLocation.pathname !== location.pathname) {
+      // Reset input text when the location changes
+      setMessageId((prev) => prev + 1)
+    }
+  }, [previousLocation, location])
```

Comment:
Can we add a hasLocationChanged var so this use effect doesn't trigger as often?

Reply:
ya